Meta/Lagom/Tools/CodeGenerators/LibUnicode/GenerateUnicodeData.cpp

```diff
@@ -844,10 +896,53 @@ static constexpr Array<Span<CodePointRange const>, @size@> @name@ { {)~~~");
     append_prop_list("s_properties"sv, "s_property_{}"sv, unicode_data.prop_list);
     append_prop_list("s_scripts"sv, "s_script_{}"sv, unicode_data.script_list);
     append_prop_list("s_script_extensions"sv, "s_script_extension_{}"sv, unicode_data.script_extensions);
+    append_prop_list("s_blocks"sv, "s_block_{}"sv, unicode_data.block_list);
     append_prop_list("s_grapheme_break_properties"sv, "s_grapheme_break_property_{}"sv, unicode_data.grapheme_break_props);
     append_prop_list("s_word_break_properties"sv, "s_word_break_property_{}"sv, unicode_data.word_break_props);
     append_prop_list("s_sentence_break_properties"sv, "s_sentence_break_property_{}"sv, unicode_data.sentence_break_props);
 
+    generator.append(R"~~~(
+struct BlockNameComparator : public CodePointRangeComparator {
+    constexpr int operator()(u32 code_point, BlockName const& name)
+    {
+        return CodePointRangeComparator::operator()(code_point, name.code_point_range);
+    }
+};
+)~~~");
+
+    generator.set("block_display_names_size", String::number(unicode_data.block_display_names.size()));
+    generator.append(R"~~~(
+static constexpr Array<BlockName, @block_display_names_size@> s_block_display_names { {
+)~~~");
+    for (auto const& block_name : unicode_data.block_display_names) {
+        generator.set("first", String::formatted("{:#x}", block_name.code_point_range.first));
+        generator.set("last", String::formatted("{:#x}", block_name.code_point_range.last));
+        generator.set("name", block_name.name);
+        generator.append(R"~~~(    { { @first@, @last@ }, "@name@"sv },
+)~~~");
+    }
+    generator.append(R"~~~(} };
+)~~~");
+
+    generator.append(R"~~~(
+Optional<StringView> code_point_block_display_name(u32 code_point)
+{
+    if (auto const* entry = binary_search(s_block_display_names, code_point, nullptr, BlockNameComparator {}))
+        return entry->display_name;
+
+    return {};
+}
+
+Vector<BlockName> block_display_names()
+{
+    Vector<BlockName> blocks;
```

Comment:
Does this need to return a `Vector`? Or could we return `Span<BlockName const>`, and then just return `s_block_display_names.span()` directly?

Reply:
Ah much nicer :ok_hand: